File: resources/php/admin2.php
<?php

require_once __DIR__ . '/bootstrap.php';

header('Content-Type: application/json; charset=utf-8');

include __DIR__ . '/config.php';

$currentUser = isset($_SESSION['username']) ? $_SESSION['username'] : 'Guest';

$conn = new mysqli($servername, $username, $password, $dbname);

if ($conn->connect_error) {
    http_response_code(500);
    echo json_encode(['error' => 'Connection failed']);
    exit;
}

include __DIR__ . '/db_init.php';

$users = [];

$result = $conn->query('SELECT userid, username, email, password FROM users ORDER BY userid ASC');
if ($result !== false) {
    while ($row = $result->fetch_assoc()) {
        $users[] = [
            'userid' => $row['userid'] ?? '',
            'username' => $row['username'] ?? '',
            'email' => $row['email'] ?? '',
            'password' => $row['password'] ?? '',
        ];
    }
}

$conn->close();

echo json_encode([
    'currentUser' => $currentUser,
    'users' => $users,
]);

File: resources/php/admin2.py
from flask import jsonify, session

from db import get_connection
from db_init import initialize_schema


def register_routes(app):
    @app.route("/resources/php/admin2.py", methods=["GET"])
    def admin2_php():
        current_user = session.get("username", "Guest")
        conn = None
        try:
            conn = get_connection()
            initialize_schema(conn)
            users = []

            with conn.cursor() as cur:
                cur.execute("SELECT userid, username, email, password FROM users ORDER BY userid ASC")
                rows = cur.fetchall() or []

            for row in rows:
                users.append(
                    {
                        "userid": row.get("userid", ""),
                        "username": row.get("username", ""),
                        "email": row.get("email", ""),
                        "password": row.get("password", ""),
                    }
                )

            return jsonify({"currentUser": current_user, "users": users})
        except Exception:
            return jsonify({"error": "Connection failed"}), 500
        finally:
            if conn is not None:
                conn.close()

